examples/store_dag.js

```diff
@@ -0,0 +1,165 @@
+    const transactions = 32;
+    const bytes = 64 * 1024 * 1024; // 64 MB
+
+    console.log('\n============================\n1. Doing authorization...');
+    await authorizeAccount(api, sudo_pair, who, transactions, bytes);
+    await new Promise(resolve => setTimeout(resolve, 7000));
+    let { nonce } = await api.query.system.account(who_pair.address);
+    console.log('Authorized!');
+
+    // 2. Store partial chunks (for example, split a big picture or video into chunks)
+    console.log('\n============================\n2. Storing data...');
+    const contents = [
+        "Hello, Bulletin remote1 - " + new Date().toString(),
+        "Hello, Bulletin remote2 - " + new Date().toString(),
+        "Hello, Bulletin remote3 - " + new Date().toString()
+    ];
+    const chunks = [];
+    for (let i = 0; i < contents.length; i++) {
+        const cid = await store(api, who_pair, contents[i], nonce.addn(i));
+        console.log(`Stored data with CID${i + 1}:`, cid);
+        chunks.push({ cid, len: contents[i].length });
+    }
+    await new Promise(resolve => setTimeout(resolve, 5000));
+
+    // 3. Read partial chunks
+    console.log('\n============================\n3. Reading partial contents...');
+    for (let i = 0; i < chunks.length; i++) {
+        const {cid, len} = chunks[i];
+        const content = await read_from_ipfs(cid);
+        console.log(`[*] CID${i + 1} ${cid} as bytes (${len} / ${content.length}), content:\n${new TextDecoder().decode(content)}`);
+    }
+
+    // 4. Store DAG file
+    console.log('\n============================\n4. DAG handling...');
+    // Construct UnixFS file DAG
+    const offset = chunks.reduce((acc, { len }) => acc + len, 0);
+    const dagFile = {
+        Data: {
+            Type: 'File',
+            filesize: offset,
+        },
+        Links: chunks.map(({ cid, len }) => ({
+            Name: '',
+            Tsize: len,
+            Hash: cid,
+        })),
+    };
+    console.log(`\n\n[DAG] Generated dagFile:\n${JSON.stringify(dagFile, null, 2)}`);
+
+    // Convert Data to bytes — the key fix:
+    const dagNode = dagPB.prepare({
+        Data: new TextEncoder().encode(JSON.stringify(dagFile.Data)),
+        Links: dagFile.Links.map(link => ({
+            Name: link.Name,
+            Tsize: link.Tsize,
+            Hash: CID.asCID(link.Hash) ?? CID.parse(link.Hash['/']) // safe fallback
+        }))
+    });
+    // Dag Encode + hash
+    const dagNodeAsBytes = dagPB.encode(dagNode);
+    const dagHash = await sha256.sha256.digest(dagNodeAsBytes);
+    const expectedRootCid = CID.createV1(dagPB.code, dagHash);
+    console.log(`[DAG] ExpectedRootCid: ${expectedRootCid}`);
+
+    // Store DAG-PB file with rootCID as and on-chain proof, can be custom pallet/state
+    console.log('\n============================\n5. Storing DAG-PB proof...');
+    await storeProof(api, who_pair, expectedRootCid);
+
+    // Store DAG-PB node in IPFS
+    const dagCid = await ipfs.dag.put(dagNode, {
```

Comment:
Ideally, this should be stored in the transaction storage as well.

Reply:
> Ideally, this should be stored in the transaction storage as well.

Yes, right :). Originally, I was storing it with `storeProof -> store(dagFile)`, but something has not been working as I expected, but I am still experimenting with this

Reply:
I finished the example with chunking - manually and/or DAG support how does that work with storing also proof for DAG: https://github.com/paritytech/polkadot-bulletin-chain/pull/85/commits/193b9282e6c254d21208b1debcb1c6fff0894e57